punchcut/src/source/glyf.rs

```diff
@@ -0,0 +1,109 @@
+/*!
+TrueType outlines loaded from the `glyf` table.
+
+*/
+
+mod math;
+mod outline;
+mod scaler;
+
+pub use outline::Outline;
+pub use scaler::Scaler;
+
+pub use read_fonts::tables::glyf::Point;
+
+/// Context for loading for TrueType glyphs.
+pub struct Context {
+    /// Unscaled points.
+    unscaled: Vec<Point>,
+    /// Original unscaled points.
+    original: Vec<Point>,
+    /// Storage for variation deltas.
+    deltas: Vec<Point>,
+}
+
+impl Default for Context {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Context {
+    /// Creates a new context.
+    pub fn new() -> Self {
+        Self {
+            unscaled: vec![],
+            original: vec![],
+            deltas: vec![],
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::{Context, Outline, Scaler};
+    use crate::{font::*, GlyphId};
```

Comment:
The code handles many scenarios we do not test, perhaps we should add more ttx's to read-fonts and make tests? - for example, variations seems to be entirely untested.

Reply:
I'd be OK deferring that to an issue that enumerates scenarios but I do think we want to test this a little more.

Reply:
I added some infra for automatically extracting the test data as mentioned above. Opened #204 to track adding ttx files to cover more/interesting cases.